plugin/cache/cache.go

```diff
@@ -203,13 +203,17 @@ func (w *ResponseWriter) WriteMsg(res *dns.Msg) error {
 	return w.ResponseWriter.WriteMsg(res)
 }
 
-func (w *ResponseWriter) set(m *dns.Msg, key uint64, mt response.Type, duration time.Duration) {
+func (w *ResponseWriter) set(m *dns.Msg, key uint64, mt response.Type, duration time.Duration, prefetch bool) {
 	// duration is expected > 0
 	// and key is valid
 	switch mt {
 	case response.NoError, response.Delegation:
 		i := newItem(m, w.now(), duration)
 		w.pcache.Add(key, i)
+		// when pre-fetching, remove the negative cache entry if it exists
```

Comment:
hmm, is this *only* need on prefetch, or maybe always?

Reply:
I don't believe so, and it would be a performance impact if we always do it.  I think due to the way things fall out, it's only needed in prefetch...

During an inline query cache retrieval, if there is an unexpired negative cache entry, we reply with that entry - no writes to cache occur, so the positive cache would remain empty.  IOW, if it was not for prefetch, there could not be an unexpired entry in both negative and positive cache.